tests/gtest/output/config.cc

```diff
@@ -0,0 +1,218 @@
+/*
+ * SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
+ * Copyright (c) 2022-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: GPL-2.0-only or BSD-2-Clause
+ */
+
+
+#include "common/def.h"
+
+/**
+ * Base class for output tests
+ */
+class output : virtual public testing::Test {
+protected:
+    std::string exec_and_capture(const std::string &cmd)
+    {
+        std::array<char, 128> buffer;
+        std::string result;
+        FILE *pipe = popen(cmd.c_str(), "r");
+        if (!pipe) {
+            throw std::runtime_error("popen() failed!");
+        }
+        while (fgets(buffer.data(), buffer.size(), pipe) != nullptr) {
+            result += buffer.data();
+        }
+        pclose(pipe);
+        return result;
+    }
+
+    void check_file(const std::string &filename, const std::vector<const char *> &expected_strings,
+        const std::vector<const char *> &unexpected_strings = {})
+    {
+        // LD_PRELOAD is already set for the gtest, it is inherited by the new process, no need to do
+        // anything
+        std::string cmd = "XLIO_USE_NEW_CONFIG=1 XLIO_CONFIG_FILE=output/" + filename + " ls 2>&1 > /dev/null";
```

Comment:
XLIO_CONFIG_FILE=output/ has an assumption here about where the tests are running from.
if the CI works, I'm fine with that :) but I'm not sure
from my chat friend It seems some change is needed:
<img width="970" height="221" alt="image" src="https://github.com/user-attachments/assets/1a1985ab-3e7f-427b-bfc4-efc1ef2914c4" />